lucene/core/src/java/org/apache/lucene/search/AbstractKnnVectorQuery.java

```diff
@@ -156,6 +196,49 @@ private TopDocs getLeafResults(
     }
   }
 
+  private DocIdSetIterator executeSeedQuery(LeafReaderContext ctx, Weight seedWeight)
+      throws IOException {
+    if (seedWeight == null) return null;
+    // Execute the seed query
+    TopScoreDocCollector seedCollector =
+        new TopScoreDocCollectorManager(
+                k /* numHits */,
+                null /* after */,
+                Integer.MAX_VALUE /* totalHitsThreshold */,
+                false /* supportsConcurrency */)
+            .newCollector();
+    final LeafReader leafReader = ctx.reader();
+    try {
+      final LeafCollector leafCollector = seedCollector.getLeafCollector(ctx);
+      if (leafCollector != null) {
+        BulkScorer scorer = seedWeight.bulkScorer(ctx);
+        if (scorer != null) {
+          scorer.score(
+              leafCollector,
+              leafReader.getLiveDocs(),
+              0 /* min */,
+              DocIdSetIterator.NO_MORE_DOCS /* max */);
+        }
+        leafCollector.finish();
+      }
+    } catch (
+        @SuppressWarnings("unused")
+        CollectionTerminatedException e) {
+    }
```

Comment:
So that `leafCollector.finish()` is called even when `CollectionTerminatedException` was caught, and because (by inspection) `getLeafCollector` on this code path does not throw that exception.
```suggestion
    final LeafReader leafReader = ctx.reader();
      final LeafCollector leafCollector = seedCollector.getLeafCollector(ctx);
      if (leafCollector != null) {
    try {
        BulkScorer scorer = seedWeight.bulkScorer(ctx);
        if (scorer != null) {
          scorer.score(
              leafCollector,
              leafReader.getLiveDocs(),
              0 /* min */,
              DocIdSetIterator.NO_MORE_DOCS /* max */);
        }
    } catch (
        @SuppressWarnings("unused")
        CollectionTerminatedException e) {
    }
        leafCollector.finish();
      }
```